src/Compilers/Core/AnalyzerDriver/AnalyzerManager.cs

```diff
@@ -142,6 +142,30 @@ internal class AnalyzerManager
         }
 
         /// <summary>
+        /// Returns true if analyzer registered a compilation start action during <see cref="DiagnosticAnalyzer.Initialize(AnalysisContext)"/>
+        /// which registered a compilation end action and at least one other analyzer action, that the end action depends upon.
+        /// </summary>
+        public async Task<bool> GetAnalyzerHasDependentCompilationEndAsync(DiagnosticAnalyzer analyzer, AnalyzerExecutor analyzerExecutor)
```

Comment:
Having the AnalyzerManager involved in per-compilation activity seems very weird to me--the AM should keep track of analyzers and initialize them, and let other components worry about transient conditions. The need to add API in order for the AM to explain what happened in compilation start actions is an example of weirdness. I'm more worried about this than about the accessibility of AnalyzerActions.

The driver simplification probably won't be part of RC. If AnalyzerManager ends up as a public type, any architectural warts we put on it might live longer than we would like.